src/index.js

```diff
@@ -1005,15 +999,31 @@ class Enmap extends Map {
     }
   }
 
-  /*
-   * INTERNAL method to verify the type of a key or property
+  /** 
+   * INTERNAL METHOD. Checks if the key is valid.
+   * Will THROW AN ERROR if the key is nil or not a string.
+   * @param {string|Array<string>} keys The key(s) to check.
+   */
+  #checkKeyOrArrayOfKeys(keys) {
+    if (!(typeof keys === 'string' || (Array.isArray(keys) && keys.every(e => typeof e === "string")))) {
+      throw new Err(
+        `Enmap requires keys to be a string. Provided: ${
+          isNil(keys) ? 'nil' : keys.constructor.name
+        }`,
+        'EnmapKeyTypeError',
+      );
+    }  
+  }
+
+  /**
+   * INTERNAL METHOD to verify a property
    * Will THROW AN ERROR on wrong type, to simplify code.
    * @param {string} key Required. The key of the element to check
    * @param {string} type Required. The javascript constructor to check
    * @param {string} path Optional. The dotProp path to the property in the object enmap.
    */
   #check(key, type, path = null) {
-    key = key.toString();
+    this.#checkKeyOrArrayOfKeys(key);
```

Comment:
Oh you did, then we can remove redundant cases

Reply:
I am a bit unsure... Some public functions use check() some don't. I feel like in general is there a lot of redundancy in the code and a consistency missing.

Reply:
Perhaps specify check() in each method call

Reply:
🤷‍♀️ I dunno. It should work how it is right now. If you want to remove the redundant checks feel free to edit it yourself.

Reply:
Well no, you made the pull request 😂, please remove the redundant checks